### PR TITLE
Update Image Handling Servlets

### PR DESCRIPTION
**Pull request by ralucatudor**
Update image handling servlets so that, when the user submits the form for image uploading, Blobstore processes the file upload and forwards the request to **PhotoHandlerServlet**, which gets the shopping results page (`jsoup Document`) for the respective photo.

Compared to the previous PR ([PR#23](https://github.com/googleinterns/step256-2020/pull/23)), states about a single request are no longer saved in the "UploadedImageInfoServlet" servlet. Now, only the PhotoHandlerServlet is called and it processes the photo, returning the results page as the response.

Note: This PR also contains commits from [PR#19](https://github.com/googleinterns/step256-2020/pull/19).

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    // the shopping query and call GoogleShoppingQuerier to return shopping results.
+
+    if (photoCategory.equals("product")) {
+      // Initialize the ProductPhotoShoppingImpl object.
+      ProductPhotoShoppingImpl productPhotoShoppingImpl = new ProductPhotoShoppingImpl();
+
+      try {
+        // Get the shopping results page as jsoup Document.
+        Document shoppingResultsDocument = productPhotoShoppingImpl.shopWithPhoto(imageBytes);
+        response.getWriter().println(shoppingResultsDocument);
```
Comment (ChangchengXiao) [inline src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java:88]: IIUC (If I Understand Correctly), line#88 doesn't need to be in the "try" block.

File: src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+   * upload a file.
+   */
+  private BlobKey getBlobKey(HttpServletRequest request, String formFileInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formFileInputElementName);
+
+    // User submitted form without selecting a file. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
```
Comment (ChangchengXiao) [inline src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java:109]: Prefer to use "Optional.empty()" instead of null.

**Review by ChangchengXiao — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java (hunk 2 above)
Comment (ralucatudor) [reply src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java:113]: What a nice suggestion, thanks, done!

File: src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java (hunk 1 above)
Comment (ralucatudor) [reply src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java]: Solved!

**Pull request merged**